app/services/stats/base_cases_report.rb

```diff
@@ -25,7 +25,7 @@ def run
       case_scope
         .where(received_date: [@period_start..@period_end])
         .order(received_date: :asc)
-        .each { |kase| @stats.add(kase) }
+        .find_each(batch_size: 100) { |kase| @stats.add(kase) }
```

Comment:
Looks good, and worth a try! Fingers crossed

Reply:
Is there a way to benchmark before and after memory use? I use the Activity Monitor on my Mac to watch the size in RAM of the ruby process, could that work locally using your test data?